### PR TITLE
fix filter

### PR DESCRIPTION
**Pull request by alpaka-alpachino**

### CODE REVIEW

**Review by crnopster — DISMISSED** [review]

File: repository/filtering.go [hunk 1]
```diff
@@ -33,15 +35,35 @@ func NewFiltersRepository(db *sql.DB) Filters {
 
 // Add adds new filter to database
 func (p *filtersRepo) Add(filter *Filter) error {
-	_, err := p.db.Exec(addQuery, filter.Name, filter.Type, filter.Configuration, filter.Reversed, filter.UserID)
+	filterID, err := uuid.NewUUID()
+	if err != nil {
+		return err
+	}
+	filter.ID = filterID
+
```
Comment (i3odja) [inline repository/filtering.go]: To my mind it is unnecessary to create new variable filterID. 
What do you think about the usage of the following:
```go
filter.ID, err = uuid.NewUUID()
if err != nil {
	return err
}
```

Comment (crnopster) [reply repository/filtering.go]: Does this example work?

Comment (alpaka-alpachino) [reply repository/filtering.go]: If we want do that in such way, we anyway need to declare var err before. So, not sure that is necessary change) WDYT?

Comment (i3odja) [reply repository/filtering.go]: Sorry, You are right, I did not notice that you deleted _, err := p.db.Exec(addQuery, filter.Name, filter.Type, filter.Configuration, filter.Reversed, filter.UserID) and put it below.

File: repository/filtering.go [hunk 2]
```diff
@@ -37,11 +39,25 @@ func (p *filtersRepo) Add(filter *Filter) error {
...
+	row := p.db.QueryRow(getOneQuery, userID, name)
+
+	err := row.Scan(filter.Name, filter.Type, filter.Configuration, filter.Reversed, filter.UserID)
+	if err != nil {
+		return nil, err
+	}
+
+	return filter, nil
+}
```
Comment (radikh) [inline repository/filtering.go]: Does the function work?

File: repository/filtering.go [hunk 3]
```diff
@@ -65,14 +81,14 @@ func (p *filtersRepo) AllByUser(userID uuid.UUID) ([]*Filter, error) {
...
+	_, err := p.db.Exec(updateQuery, filter.Type, filter.Configuration, filter.Reversed, filter.UserID, filter.Name)
 	return err
 }
 
-// Delete deletes filter by name for relevant user
-func (p *filtersRepo) Delete(userID uuid.UUID, filterName string) error {
-	_, err := p.db.Exec(deleteQuery, userID, filterName)
+// Delete deletes filter by id for relevant user
+func (p *filtersRepo) Delete(userID uuid.UUID, name string) error {
+	_, err := p.db.Exec(deleteQuery, userID, name)
```
Comment (radikh) [inline repository/filtering.go]: filter name - was a good parameter name.

File: repository/filtering.go (hunk 2 above)
Comment (alpaka-alpachino) [reply repository/filtering.go]: fixed)

File: repository/filtering.go (hunk 3 above)
Comment (alpaka-alpachino) [reply repository/filtering.go]: fixed)

**Review by crnopster — DISMISSED** [review]

**Review by maksymbilokur — DISMISSED** [review]

**Review by Rodion-Shytykov — COMMENTED** [review]
I guess some tests should be changed

**Review by crnopster — DISMISSED** [review]

**Review by crnopster — DISMISSED** [review]

**Review by crnopster — APPROVED** [review]

**Review by radikh — APPROVED** [review]

**Review by maksymbilokur — APPROVED** [review]

**Pull request merged**